akka-actor/src/main/scala/akka/actor/ActorSystem.scala

```diff
@@ -109,6 +207,26 @@ object ActorSystem {
   def apply(name: String): ActorSystem = apply(name, None, None, None)
 
   /**
+   * Scala API: Creates a new actor system with the specified name and settings
+   * The core actor system settings are defined in [[BootstrapSetup]]
+   */
+  def apply(name: String, setup: ActorSystemSetup): ActorSystem = {
```

Comment:
I thought about it for a bit but ended up with it being a property of the actor system rather than a setup thingie. I can probably be convinced though.